db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java

```diff
@@ -1290,6 +1290,7 @@ public static DataValueDescriptor getNullDVDWithUCS_BASICcollation(
             case StoredFormatIds.SQL_TIMESTAMP_ID: return new SQLTimestamp();
             case StoredFormatIds.SQL_TINYINT_ID: return new SQLTinyint();
             case StoredFormatIds.SQL_VARCHAR_ID: return new SQLVarchar();
+            case StoredFormatIds.SQL_VARCHAR_DB2_COMPATIBLE_ID: return new SQLVarcharDB2Compatible();
```

Comment:
Do we need to add `case StoredFormatIds.SQL_VARCHAR_DB2_COMPATIBLE_ID` for getNullDVD() also?

Reply:
I don't think so.  The only place EngineUtils.getFormatIds is called is in SparkValueRowSerializer.write, and the only place this is consumed is in SparkValueRowSerializer.read:

>                 formatIds[i] = input.readInt(true);
>                 rowTemplate[i] = getDVD(formatIds[i]);

Also, getNullDVD deals with different typeFormatIds (the ones that don't start with "SQL_").

Reply:
I see, thanks Mark!